src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_BuildSynchronization.cs

```diff
@@ -66,13 +66,7 @@ internal partial class DiagnosticIncrementalAnalyzer
                         var result = GetResultOrEmpty(newResult, stateSet.Analyzer, project.Id, VersionStamp.Default);
 
                         // Save into in-memory cache.
-                        await state.SaveInMemoryCacheAsync(project, result).ConfigureAwait(false);
-
-                        // Save into persistent storage on a separate post-build and post error list refresh task queue.
-                        _ = postBuildAndErrorListRefreshTaskQueue.ScheduleTask(
-                                nameof(SynchronizeWithBuildAsync),
-                                () => state.SaveAsync(PersistentStorageService, project, result),
-                                cancellationToken);
+                        state.SaveToInMemoryStorage(project, result);
```

Comment:
we were both saving to in-memory cache and saving to persistence layer.  now that the latter is gone, this is jsut a single call with no task queueing.